src/modules/Hosts/HostsUILib/Helpers/DuplicateService.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using HostsUILib.Models;
+using HostsUILib.Settings;
+using Microsoft.UI.Dispatching;
+
+namespace HostsUILib.Helpers
+{
+    public class DuplicateService : IDuplicateService, IDisposable
+    {
+        private record struct Check(string Address, string[] Hosts);
+
+        private readonly IUserSettings _userSettings;
+        private readonly DispatcherQueue _dispatcherQueue;
+        private readonly Queue<Check> _checkQueue;
+        private readonly ManualResetEvent _checkEvent;
+        private readonly Thread _queueThread;
+
+        private readonly string[] _loopbackAddresses =
+        {
+            "0.0.0.0",
+            "::0",
+            "0:0:0:0:0:0:0:0",
```

Comment:
Is it possible that a user types the long version with 4 zeros per section? Is a regex better here to be more flexible regarding the input (short vs long)?

Reply:
RFC says short version: `::`

_https://www.rfc-editor.org/rfc/rfc4291#section-2.5.2_